target/linux/realtek/dts/rtl8380_netgear_gigabit.dtsi

```diff
@@ -0,0 +1,141 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include "rtl838x.dtsi"
+
+#include <dt-bindings/input/input.h>
+#include <dt-bindings/gpio/gpio.h>
+
+/ {
+	compatible = "realtek,rtl838x-soc";
+
+	chosen {
+		bootargs = "console=ttyS0,115200";
+	};
+
+	memory@0 {
+		device_type = "memory";
+		reg = <0x00000000 0x08000000>;
+	};
+
+	keys {
+		compatible = "gpio-keys-polled";
+		poll-interval = <20>;
+
+		mode {
+			label = "reset";
+			gpios = <&gpio0 24 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RESTART>;
+		};
+	};
+};
+
+&gpio0 {
+	indirect-access-bus-id = <0>;
+};
+
+&spi0 {
+	status = "okay";
+
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <50000000>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "u-boot";
+				reg = <0x00000000 0x000e0000>;
+				read-only;
+			};
+
+			partition@e0000 {
+				label = "u-boot-env";
+				reg = <0x000e0000 0x00010000>;
+				read-only;
+			};
+
+			partition@f0000 {
+				label = "sysinfo";
+				reg = <0x000f0000 0x00010000>;
+				read-only;
+			};
+
+			partition@100000{
+				label = "jffs2_cfg";
+				reg = <0x00100000 0x00100000>;
+				read-only;
+			};
+
+			partition@200000{
+				label = "jffs2_log";
+				reg = <0x00200000 0x00100000>;
+				read-only;
+			};
+
+			partition@300000{
+				label = "firmware";
+				compatible = "openwrt,uimage", "denx,uimage";
+				openwrt,ih-magic = <0x4e474520>;
+				openwrt,ih-type = <IH_TYPE_KERNEL>;
```

Comment:
Looks like ih-type is what causes the syntax error. And that's also the change you introduce which wasn't there before.

Reply:
It's due to missing #include <dt-bindings/mtd/partitions/uimage.h>

Reply:
I didn't test this, but I think this breaks due to a missing include of "dt-bindings/mtd/partitions/uimage.h", introduced in dd5ac0546ca3f80de10db309e57090cf809b1b72

Reply:
yes, all these properties are meant only for overriding defaults.  No need to repeat default values in the dts